src/group_rings.jl

```diff
@@ -4,22 +4,30 @@ Base.adjoint(X::AlgebraElement) = StarAlgebras.star(X)
 Base.copy(X::AlgebraElement) =
     AlgebraElement(copy(StarAlgebras.coeffs(X)), parent(X))
 StarAlgebras.star(g::Groups.GroupElement) = inv(g)
+using AbstractAlgebra
+StarAlgebras.star(M::AbstractAlgebra.Generic.MatAlgElem{Int64}) = inv(M)
 
 # Group ring with the basis given by the prescribed support
-function group_ring(G, half_basis, star_multiplication = false)
+function group_ring(G, half_basis; star_multiplication = false, additive_only = false)
```

Comment:
why do you need `additive_only`?
If you call
```julia
    tmstr = StarAlgebras.CachedMTable{star_multiplication}(
        group_ring_basis,
        table_size = (length(star_closed_support), length(star_closed_support)),
    )
    RG = StarAlgebra(G, group_ring_basis, tmstr)
```
then the table is empty (no precomputation is happening), so it should be as fast as 
`RG = StarAlgebra(G, group_ring_basis)` (save the allocation time, which could be cut in more than half if you use `UInt32` for `group_ring_basis`.

If you really want `TrivialMStructure` there construct it explicitly with the correct size:
```julia
mstr = let elements= star_closed_support, tw = star_multiplication
    basis = StarAlgebras.Basis{UInt32}(elements)
    StarAlgebras.TrivialMStructure{tw}(basis)
end
```